base/inference.jl

```diff
@@ -542,7 +542,38 @@ add_tfunc(===, 2, 2,
         end
         return Bool
     end)
-add_tfunc(isdefined, 1, IInf, (args...)->Bool)
+function isdefined_tfunc(args...)
+    arg1 = args[1]
+    if isa(arg1, Const)
+        a1 = typeof(arg1.val)
+    else
+        a1 = widenconst(arg1)
+    end
+    if isType(a1)
+        return Bool
+    end
+    if isleaftype(a1)
+        if a1 <: Array # TODO
+        elseif a1 === Module # TODO
+        elseif length(args) == 2 && isa(args[2], Const)
+            n = nfields(a1)
+            val = args[2].val
+            idx::Int = 0
+            if isa(val, Symbol)
+                idx = fieldindex(a1, val, false)
+            elseif isa(val, Int)
+                idx = val::Int
+            end
+            if 1 <= idx <= a1.ninitialized
+                return Const(true)
+            elseif idx <= 0 || idx > n
+                return Const(false)
+            end
+        end
+    end
+    Bool
+end
+add_tfunc(isdefined, 1, IInf, isdefined_tfunc)
```

Comment:
upper end is actually 2, not `IInf` (noticed this on my PR https://github.com/JuliaLang/julia/pull/22281/files#diff-c0d983631c937ee5acfd0137cd7332deR545). (although when we remove the deprecations after v0.7, this'll just become constant at 2 args anyways)